src/main/java/gift/service/CategoryService.java

```diff
@@ -20,8 +22,20 @@ public CategoryService(CategoryRepository categoryRepository) {
     public List<CategoryResponse> getAllCategories() {
         List<Category> categories = categoryRepository.findAll();
         List<CategoryResponse> responses = categories.stream()
-                                        .map(cat -> CategoryResponse.fromEntity(cat))
-                                        .toList();
+            .map(cat -> CategoryResponse.fromEntity(cat))
+            .toList();
         return responses;
     }
+
+    public CategoryResponse addCategory(CategoryRequest newCategory) {
+        categoryRepository.findByName(newCategory.getName())
+            .ifPresent(cat -> {
+                throw new DuplicateCategoryNameException();
+            });
+        Category category = newCategory.toEntity();
+        categoryRepository.save(category);
+
+        CategoryResponse response = CategoryResponse.fromEntity(category);
+        return response;
```

Comment:
새롭게 저장된 객체를 불러와서 Response로 넘기는 방법이 더 안전할 것 같아요 :)

```suggestion
Category category = newCategory.toEntity();
Category savedCategory = categoryRepository.save(category);
return CategoryResponse.fromEntity(savedCategory);
```